src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/blocktransactions/FullBlockTransactionsProcessor.scala

```diff
@@ -22,8 +21,7 @@ trait FullnodeBlockTransactionsProcessor extends BlockTransactionsProcessor with
     historyStorage.modifierById(txs.headerId) match {
       case Some(header: Header) =>
         historyStorage.modifierById(header.ADProofsId) match {
-          case _ if !header.isGenesis && bestFullBlockIdOpt.isEmpty =>
-            //TODO light mode when start from different block ?
+          case _ if bestFullBlockIdOpt.isEmpty && !isValidFirstFullBlock(header) =>
```

Comment:
if we introduce isEmptyHistory method, then we could use it here instead of `bestFullBlockIdOpt.isEmpty`

Reply:
History may contain headers (so it's non-empty) while no full blocks yet